bytestring_splitter/__init__.py

```diff
@@ -534,13 +548,24 @@ class StructureChecksumMixin(HeaderMetaDataMixinBase):
     HEADER_LENGTH = 4
     METADATA_TAG = 'checksum'
 
-    @classmethod
-    def _deserialize_metadata(cls, data_bytes):
-        return data_bytes.decode('ascii')
-
-    @classmethod
-    def _serialize_metadata(cls, value):
-        return b'dave'
+    class InvalidBytestringException(BaseException):
+        pass
+
+    def generate_checksum(self, *args, **kwargs):
+        hash = hashlib.sha256()
```

Comment:
After all that hullaballo, you went with sha256?  This is way overkill (in terms of cpu use).

Reply:
I picked that because I know it's widely supported and it's in python.Hashlib.  I'm not up to speed on which to use in a situation like this.  MD5?  Sha1?  Preferably one that exists in Javascript and the Python standard lib.

Reply:
Hmm Sha1 isn't much faster.

Reply:
Ok for the record.
for 10,000,000 iterations on my decent laptop:
* sha256:  14.09s
* sha1: 13.74s
 * md5: 13.60s

That seems not to be worth trifling about but it's possible that someday, some situation of great consequence might be resolved more favorably when someone reads in 10,000,000 bytestrings and finishes .49 seconds sooner.

MD5 it is!

Reply:
Also made it very easy to override